x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -597,16 +591,18 @@ export const MLInferenceLogic = kea<
               destination_field: destinationField,
               model_id: modelId,
               source_field: sourceField,
+              field_mappings: fieldMappings,
             } = pipelineParams;
 
+            const missingSourceFields =
```

Comment:
Good catch, I'll add a test to the missing source fields use case. For the others I'm not planning to because
- ELSER filtering is being removed in this very PR
- @julianrosado and I discussed that we should filter pipelines from the list that are already attached to the current index: the list will be visible in the ML Inference Pipelines panel, and it doesn't make sense to list a non-selectable item in a select dropdown. (I'll file an issue for this)